src/components/Form.js

```diff
@@ -303,7 +304,8 @@ function Form(props) {
 
                 // We want to initialize the input value if it's undefined
                 if (_.isUndefined(inputValues[inputID])) {
-                    inputValues[inputID] = _.isBoolean(defaultValue) ? defaultValue : defaultValue || '';
+                    // eslint-disable-next-line es/no-nullish-coalescing-operators
+                    inputValues[inputID] = defaultValue ?? '';
```

Comment:
I made this change because it's more simpler and does the same thing as before.

Reply:
<img width="424" alt="image" src="https://github.com/Expensify/App/assets/38015950/f34dafd9-17d2-48eb-8c34-6ae93cb13412">

@pac-guerreiro @allroundexperts Shall we revert this line to avoid breaking other forms?

Reply:
<img width="436" alt="image" src="https://github.com/Expensify/App/assets/38015950/77a6faf9-34e0-4dc9-9d6b-e509bff5775f">

although @allroundexperts is this the case you were referring to?